src/linux_ip_socket/linux_ip_socket.cpp

```diff
@@ -0,0 +1,336 @@
+{
+    m_ip_device_bus_id = bus_id;
+    m_ip_device_id = device_id;
+    m_ip_device_configuration = device_configuration;
+    m_ip_remote_device_configuration = remote_device_configuration;
+    m_thread.start(&taste::LinuxIpSocketPoll, this);
+}
+
+void
+linux_ip_socket_private_data::driver_poll()
+{
+    prepare_listen_socket();
+
+    const int descriptors_table_size = DRIVER_MAX_CONNECTIONS + 1;
+    struct pollfd descriptors_table[descriptors_table_size];
+    int descriptors_table_count = 0;
+    descriptors_table[0].fd = m_listen_sockfd;
+    descriptors_table[0].events = POLLIN;
+    ++descriptors_table_count;
+
+    while(true) {
+        const int poll_result = ::poll(descriptors_table, descriptors_table_count, -1);
+        if(poll_result == -1) {
+            std::cerr << "poll() returned an error: " << strerror(errno) << std::endl;
+            std::cerr << "aborting." << std::endl;
+            abort();
+        }
+
+        for(int descriptor_index = 0; descriptor_index < descriptors_table_count; ++descriptor_index) {
+            if(descriptors_table[descriptor_index].revents & POLLIN) {
+                if(descriptor_index == 0) {
+                    accept_connection(descriptors_table, descriptors_table_count);
+                } else {
+                    read_data_or_disconnect(descriptor_index, descriptors_table, descriptors_table_count);
+                }
+            }
+        }
+    }
+}
+
+void
+linux_ip_socket_private_data::driver_send(const uint8_t* const data, const size_t length)
+{
+    const int sockfd = connect_to_remote_driver();
+    if(sockfd == INVALID_SOCKET_ID) {
+        return;
+    }
+
+    size_t send_buffer_index = 0;
+
+    m_send_buffer[0] = START_BYTE;
+    ++send_buffer_index;
+
+    size_t index = 0;
+    bool escape = false;
+    while(index < length) {
+        if(escape) {
+            m_send_buffer[send_buffer_index] = data[index];
+            ++send_buffer_index;
+            ++index;
+            escape = false;
+        } else if(data[index] == START_BYTE) {
+            m_send_buffer[send_buffer_index] = ESCAPE_BYTE;
+            ++send_buffer_index;
+            escape = true;
+        } else if(data[index] == ESCAPE_BYTE) {
+            m_send_buffer[send_buffer_index] = ESCAPE_BYTE;
+            ++send_buffer_index;
+            escape = true;
+        } else if(data[index] == STOP_BYTE) {
+            m_send_buffer[send_buffer_index] = ESCAPE_BYTE;
+            ++send_buffer_index;
+            escape = true;
+        } else {
+            m_send_buffer[send_buffer_index] = data[index];
+            ++send_buffer_index;
+            ++index;
+        }
+
+        if(send_buffer_index >= DRIVER_SEND_BUFFER_SIZE) {
+            send_packet(sockfd, DRIVER_SEND_BUFFER_SIZE);
+            send_buffer_index = 0;
+        }
+    }
+
+    m_send_buffer[send_buffer_index] = STOP_BYTE;
+    ++send_buffer_index;
+    send_packet(sockfd, send_buffer_index);
+    send_buffer_index = 0;
```

Comment:
Why is this local variable zeroed without further use? Doesn't it produce a warning during compilation?

Reply:
Changed. There wasn't compiler warning.